exercicioNum02/bio-name-remove-update.ts

```diff
@@ -0,0 +1,152 @@
+}
+
+/**
+ * Retorna o nome de uma pessoa com base no ID.
+ *
+ * @param id - O ID da pessoa
+ * @returns O nome da pessoa ou indefinido se não encontrada
+ */
+function getNameById(id: number): Descricao {
+    return lista.find(objectList => objectList.id === id)?.name
+}
+
+/**
+ * Remove uma pessoa da lista com base no ID.
+ *
+ * @param id - O ID da pessoa a ser removida
+ * @returns Verdadeiro se a pessoa foi encontrada e removida, falso caso contrário
+ */
+function removeItemById(id: number): boolean {
+    const index = lista.findIndex(objectList => objectList.id === id)
+    if (index !== -1) {
+        lista.splice(index, 1)
+        return true
+    }
+    return false
+}
+
+/**
+ * Solicita ao usuário o ID da pessoa e retorna a biografia e o nome.
+ *
+ * @returns Um objeto contendo a biografia e o nome da pessoa
+ */
+function promptUserForId(): { bioUser: string, nameUser: string } {
+    let bioUser = ""
+    let nameUser = ""
+
+    const inputUser = prompt("Insira o ID da pessoa:")
+    const id = Number(inputUser)
+
+    if (!isNaN(id)) {
+        const bio = getBioById(id)
+        const name = getNameById(id)
+        if (bio && name) {
+            bioUser = `Biografia: ${bio}`
+            nameUser = `${name}`
+        } else {
+            bioUser = "ID não encontrado."
+            nameUser = "Nome não encontrado."
+        }
+    } else {
+        console.log("Entrada inválida. Por favor, insira um número.")
+    }
+    return { nameUser, bioUser }
+}
+/**
+ * Atualiza o nome e a biografia de uma pessoa na lista com base no ID.
+ *
+ * @param id - O ID da pessoa a ser atualizada
+ * @param newName - O novo nome da pessoa (opcional)
+ * @param newBio - A nova biografia da pessoa (opcional)
+ * @returns Verdadeiro se a pessoa foi encontrada e atualizada, falso caso contrário
+ */
+function updatePersonById(id: number, newName?: string, newBio?: string): boolean {
+    const person = lista.find(objectList => objectList.id === id)
+    if (person) {
+        if (newName && newBio) {
+            person.name = newName
+            person.bio = newBio
+        }
+        return true
+    } else {
+        return false
+    }
+}
+/**
+ * Solicita ao usuário o ID da pessoa a ser atualizada e os novos dados.
+ * Atualiza a pessoa na lista se o ID for válido.
+ */
+function UpdateUser(): void {
+    const inputUser = prompt("Insira o ID da pessoa que deseja atualizar:")
+    const id = Number(inputUser)
+
+    if (!Number.isNaN(id)) {
+        const newName = prompt("Insira o novo nome (deixe em branco para não alterar):")
+        const newBio = prompt("Insira a nova biografia (deixe em branco para não alterar):")
+
+        const wasUpdated = updatePersonById(id, newName || undefined, newBio || undefined)
+        if (wasUpdated) {
+            console.log(`Item com ID ${id} foi atualizado.`)
+        } else {
+            console.log(`Item com ID ${id} não encontrado.`)
+        }
+    } else {
```

Comment:
Vou deixar como um bônus para ti aqui, pesquisar sobre Early Return Pattern, é uma prática comum que desenvolvedores utilizam e da um upgrade visual no código. :)

https://www.lucaspaganini.com/pt/academy/early-returns-code-patterns-in-javascript